### PR TITLE
Refactor Association Profile and several other fixes

### PR DESCRIPTION
**Pull request by oskar-codes**
The main objective of this pull request is to fix an issue that came with #104, where a backend request was made directly from a Composable's scope in `AssociationProfile`, which thus made Firestore requests at every composable refresh.
This issue is now fixed, along with a refactor of the `AssociationViewModel`. To keep consistency accross the project, the same refactor was reflected in in `UserProfile`.
The most notable change of the refactor is to remove the use of the string parameter `{uid}` of the association profile route. Instead, the association is stored in the `AssociationViewModel`. This was decided after discussing with @armouldr, who is currently implementing the same process for the `EventDetails`.

### CODE REVIEW

File: app/src/androidTest/java/com/android/unio/ui/association/AssociationProfileTest.kt [hunk 1]
```diff
@@ -180,67 +172,20 @@ class AssociationProfileTest {
   @Test
   fun testAssociationProfileGoodId() {
     composeTestRule.setContent {
-      AssociationProfileScreen(navigationAction, "1", associationViewModel, userViewModel)
+      AssociationProfileScaffold(
```
Comment (armouldr) [inline app/src/androidTest/java/com/android/unio/ui/association/AssociationProfileTest.kt:175]: great use of the Mock infrastructure oskar !

File: app/src/androidTest/java/com/android/unio/ui/explore/ExploreScreenTest.kt [hunk 2]
```diff
@@ -149,8 +150,8 @@ class ExploreScreenTest {
     sortedByCategoryAssociations.forEach { (_, associations) ->
       associations.forEach {
         composeTestRule.onNodeWithTag("associationItem_${it.name}").performClick()
-        verify(navigationAction).navigateTo(Screen.withParams(Screen.ASSOCIATION_PROFILE, it.uid))
       }
     }
+    verify(navigationAction, atLeastOnce()).navigateTo(Screen.ASSOCIATION_PROFILE)
```
Comment (armouldr) [inline app/src/androidTest/java/com/android/unio/ui/explore/ExploreScreenTest.kt:155]: Oh nice ! I didn'tknow about the atLeastOnce functionality :)

File: app/src/main/java/com/android/unio/model/association/AssociationViewModel.kt [hunk 3]
```diff
@@ -98,4 +101,12 @@ constructor(
           return it
         } ?: return null
   }
+
+  fun selectAssociation(associationId: String) {
+    _selectedAssociation.value =
+        findAssociationById(associationId).also {
+          it?.events?.requestAll()
+          it?.members?.requestAll()
```
Comment (armouldr) [inline app/src/main/java/com/android/unio/model/association/AssociationViewModel.kt:109]: Great use of the kotlin semantics :+1:

File: app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt [hunk 4]
```diff
@@ -340,13 +326,17 @@ private fun AssociationProfileEvents(
...
         }
     Spacer(modifier = Modifier.size(11.dp))
-    OutlinedButton(
-        onClick = { isSeeMoreClicked = true },
-        modifier = Modifier.padding(horizontal = 28.dp).testTag("AssociationSeeMoreButton")) {
-          Icon(Icons.AutoMirrored.Filled.ArrowForward, contentDescription = "See more")
-          Spacer(Modifier.width(2.dp))
-          Text(context.getString(R.string.association_see_more))
-        }
+    if (events.size > 1) {
```
Comment (armouldr) [inline app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt:329]: Good catch !

**Review by armouldr — APPROVED** [review]
Useful refactoring that will make the project more maintanable in the long run :)

**Review by Aurelien9Code — APPROVED** [review]
Great job @oskar-codes !

**Comment by oskar-codes** [comment]
This pull request introduced an issue on main. The `UserProfileScreenContent` composable does not properly navigate to joined associations. This is because before navigating to a new screen, you need to select the association in the `AssociationViewModel`.

**Pull request merged**